### PR TITLE
Dependency updates, Maven plugin updates, refactorings and cleanups

### PR DESCRIPTION
**Pull request by dheid**
Hi @jamesward !

Again some dependency updates, refactorings, Maven plugin updates and stuff to the WebJars locator

Perhaps the most important change is the usage of Mustache template engine to generate the setup JavaScript snippet using a template file.

Best regards

Daniel

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -41,29 +42,51 @@
...
+        <system>GitHub Actions</system>
+        <url>https://github.com/webjars/webjars-locator/actions</url>
+    </ciManagement>
+
     <dependencies>
         <dependency>
             <groupId>org.webjars</groupId>
             <artifactId>webjars-locator-core</artifactId>
-            <version>0.48</version>
+            <version>0.48</version> <!-- please replace with version 0.49 after release -->
```
Comment (jamesward) [inline pom.xml]:
```suggestion
            <version>0.50</version>
```

File: pom.xml [hunk 2]
```diff
@@ -161,7 +260,7 @@
             <plugin>
                 <groupId>org.sonatype.plugins</groupId>
                 <artifactId>nexus-staging-maven-plugin</artifactId>
-                <version>1.6.6</version>
+                <version>1.6.11</version>
```
Comment (jamesward) [inline pom.xml]:
```suggestion
                <version>1.6.8</version>
```

**Comment by jamesward** [comment]
Thanks for this!  My only concern with this is the runtime templating having a performance impact.  I'd rather switch this to a compile time template.  Maybe there is a Maven plugin to compile the mustache template at compile time?

**Comment by dheid** [comment]
Mustache is extremly fast and the template is cached once it was requested for a specific path. Sadly you cannot precompile the template, since most of it will be build according to the WebJar classpath elements. So from my understanding we cannot compile it in advance. It won't be much slower than the previous string concatenation.

See also: https://github.com/mbosecke/template-benchmark

**Comment by jamesward** [comment]
Ah cool. Since it gets cached, that is sufficient reason. Thanks for the details!

**Comment by jamesward** [comment]
I'm releasing `0.43` now.

**Pull request merged**